src/test/scala/com/github/music/of/the/ainur/quenya/Test.scala

```diff
@@ -50,8 +51,55 @@ class Test extends FunSuite with BeforeAndAfter {
     assert(diff == 0)
   }
 
+
+  val xmlDf1 = xmlToDf("<foo><bar>1</bar><bar>1</bar></foo>")
+  val xmlDf2 = xmlToDf("<foo><bar>1</bar></foo>")
+
+
+
+  val xmlDsl = quenyaDsl.compile("""
+      |bar@ba
+      | ba$bar:LongType""".stripMargin)
+
+  val xmlDslDf1 = quenyaDsl.execute(xmlDsl, xmlDf1)
+  val xmlDslDf2 = quenyaDsl.execute(xmlDsl, xmlDf2)
+  val xmlDslDf1Count = xmlDslDf1.count()
+  val xmlDslDf2Count = xmlDslDf2.count()
+
+
+  // test to check the number of elements in the df from first xml string are exactly 2
+  test("number of records should be 2"){
+    assert(xmlDslDf1Count == 2)
+  }
+
+  // test to check the number of elements in the df from second xml string are exactly 1
+  test("number of records should be 1"){
+    assert(xmlDslDf2Count == 1)
+  }
+
+  val checkXml1 = checkData(xmlDslDf1)
+  val checkXml2 = checkData(xmlDslDf2)
+  // test to check the data of df from first xml
+  test("checking data for first xml"){
+    assert(checkXml1)
+  }
+
+  // test to check the data of df from second xml
+  test("checking data for second xml"){
+    assert(checkXml2)
+  }
   after {
     spark.stop()
   }
 
+  // returns true if all the elements in the bar column of df are equal to 1
+  def checkData(df:DataFrame): Boolean = {
+    df.select("bar").as[Long].collect().forall(i=>i==1)
```

Comment:
Hi Daniel, Since all the elements in column bar are 1, I am making sure of that with the above method. I don't know if this is the right way to do it. Please suggest better method if any. Thank you.